core/templates/dev/head/pages/exploration_editor/editor_tab/StateSolution.js

```diff
@@ -0,0 +1,356 @@
+                  id: choice.val,
+                  val: choice.label,
+                  clickedRegions: choice.label
+                };
+              }));
+        } else {
+          $rootScope.ruleDescriptionChoices = (
+            responsesService.getAnswerChoices().map(
+              function (choice) {
+                return {
+                  id: choice.val,
+                  val: choice.label
+                };
+              }));
+        }
+        if (interactionId === 'ItemSelectionInput') {
+          $rootScope.objectType = 'SetOfHtmlString';
+        } else if (interactionId === 'ImageClickInput') {
+          $rootScope.objectType = 'RegionSelect'
+        } else {
+          $rootScope.objectType = 'select';
+        }
+      }
+    });
+
+    $scope.getSolutionSummary = function() {
+      var solution = stateSolutionService.displayed;
+      var isExclusiveAnswer = (
+        solution.answerIsExclusive ? 'Only' : 'One');
+      var correctAnswer = '';
+      if (typeof solution.correctAnswer === 'string') {
+        correctAnswer = (
+          $filter('convertToPlainText')(solution.correctAnswer));
+      } else if (Array.isArray(
+          solution.correctAnswer)) {
+        correctAnswer = (
+          '[' + solution.correctAnswer + ']');
+      } else {
+        correctAnswer = (
+          ($scope.objectType === 'Graph') ? 'Graph Object' : 'Object');
+      }
+      var explanation = (
+        $filter('convertToPlainText')(solution.explanation));
+      return (
+        '[' + isExclusiveAnswer + ' solution is ' + correctAnswer + '] ' +
+        explanation);
+    };
+
+    // This returns false if the current interaction ID is null.
+    $scope.isCurrentInteractionLinear = function() {
+      var interactionId = stateInteractionIdService.savedMemento;
+      return interactionId && INTERACTION_SPECS[interactionId].is_linear;
+    };
+
+    $scope.openAddSolutionModal = function() {
+      alertsService.clearWarnings();
+      $rootScope.$broadcast('externalSave');
+      $scope.isActive = false;
+
+      $modal.open({
+        templateUrl: 'modals/addSolution',
+        backdrop: 'static',
+        controller: [
+          '$scope', '$rootScope', '$modalInstance', 'editorContextService',
+          function($scope, $rootScope, $modalInstance, editorContextService) {
+            $scope.EXPLANATION_FORM_SCHEMA = {
+              type: 'html',
+              ui_config: {}
+            };
+
+            var interactionId = stateInteractionIdService.savedMemento;
+            $scope.INTERACTION_SPECS = INTERACTION_SPECS;
+
+            $scope.tmpSolution = {};
+            $scope.tmpSolution.answerIsExclusive = (
+              $scope.answerIsExclusive === 'true');
+            $scope.tmpSolution.correctAnswer = '';
+            $scope.tmpSolution.explanation = '';
+
+            $scope.addSolutionForm = {};
+            if ($rootScope.objectType === 'Graph') {
+              $scope.tmpSolution.correctAnswer = {
```

Comment:
I'm concerned about all the manual construction here. The answer needs to be exactly the same as what gets submitted via the learner interface, and both routes should therefore make use of the same answer construction pathways and form interfaces. Otherwise, if you change one, the other won't get changed.

In general, try to reuse as much as possible from the learner view side of things.